### ISSUE TITLE
If jwksUri contains a port number, DNS resolution fails

### ISSUE THREAD

**Issue by gyszalai** [auth0/node-jwks-rsa#234]
### Describe the problem

JwksClient uses hostname:port for hostname resolution so if the jwksUri contains a port number, hostname resolution fails.
The cause of the problem if that the request wrapper uses the `host` property from the parsed URL instead of the `hostname` property.

### What was the expected behavior?

Hostname resolution should work even if the jwksUri contains a port number

### Reproduction

```typescript
import JwksClient from 'jwks-rsa'

const jwksClient = JwksClient({
  jwksUri: 'http://myhost:8888/jwks'
})

async function getKey(keyId: string) => {
  const key = await jwksClient.getSigningKey(keyId)
  return { cert: key.getPublicKey() }
}

const key = await getKey('1)
```

### Environment

- **Version of this library used:** 2.0.1
- **Which framework are you using, if applicable:** -
- **Other modules/plugins/libraries that might be involved:** -
- **Any other relevant information you think would be useful:** -

**Comment by carboneater** [comment]
Looks like we have a proposed solution in #233

**Comment by gyszalai** [comment]
Sorry, my fault. Didn't check PRs before filing the issue.

**Comment by jonjrodriguez** [comment]
Having the same issue when trying to use localstack.

**Comment by davidpatrick** [comment]
Fixed in #233
